tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/TraceAnnotations/TraceAnnotationInfoFactory.cs

```diff
@@ -0,0 +1,26 @@
+// <copyright file="TraceAnnotationInfoFactory.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Reflection;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations
+{
+    internal static class TraceAnnotationInfoFactory
+    {
+        public static TraceAnnotationInfo Create(MethodBase? method)
+        {
+            if (method is null)
+            {
+                return TraceAnnotationInfo.Default;
+            }
+            else
+            {
+                return new TraceAnnotationInfo(resourceName: method.Name);
```

Comment:
I was expecting the support for custom attributes here, I suspose this will be included in a future PR.

Reply:
Yes, I'll include that in a separate PR